packages/searchkit-client/src/withSearchkitRouting.tsx

```diff
@@ -10,21 +10,30 @@ import {
 } from './searchkit'
 import type { SearchState } from './searchkit'
 
-export const routeStateEqual = (a, b) =>
-  Object.entries(a).toString() === Object.entries(b).toString()
+const sanitiseRouteState = (routeState) => {
+  const intKeys = ['size', 'from']
```

Comment:
also there were situations where the size and from values wernt being casted to int. I make sure it is before equality check between states.